src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -306,27 +299,9 @@ def teardown(self) -> None:
     def on_save_checkpoint(self) -> Dict:
         state_dict = super().on_save_checkpoint()
         state_dict["_batches_that_stepped"] = self._batches_that_stepped
-
-        trainer = self._trainer
-        if (
-            trainer is not None
-            and trainer.state._fault_tolerant_mode.is_enabled
-            and trainer.train_dataloader is not None
-            and not self._num_completed_batches_reached()  # did not finish
-            # TODO: fault-tolerance requires a minimum number of batches so probably should be > 0
-            and self.batch_progress.current.ready  # did start
-        ):
-            assert isinstance(trainer.train_dataloader, CombinedLoader)
-            loader: CombinedLoader = trainer.train_dataloader
-            state = loader.state_dict(has_completed=self._has_completed())
-            if state:
-                state_dict["dataloader_state_dict"] = _collect_states_on_rank_zero_over_collection(state)
-
         return state_dict
 
     def on_load_checkpoint(self, state_dict: Dict) -> None:
-        # cache the dataloader state dict until the dataloader objects are available
-        self._dataloader_state_dict = state_dict.get("dataloader_state_dict", {})
         self._batches_that_stepped = state_dict.get("_batches_that_stepped", 0)
```

Comment:
@carmocca should we implement a migration that drops these keys? The feature was experimental, we could also just leave the keys in the checkpoint unchanged.

Reply:
What would be the benefit? Loading should work without issues as it is. They would just be unused.

Reply:
Just for your awareness. They would be unused yes. The user could do something with it as well.